core/loadBalancer/src/whisk/core/loadBalancer/ActivationThrottle.scala

```diff
@@ -17,149 +17,146 @@
 package whisk.core.loadBalancer;
 
 import scala.BigInt
-import scala.collection.concurrent.TrieMap
-import scala.math.BigInt.int2bigInt
-import scala.util.Try
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
 
-import spray.json.JsNumber
+import akka.actor.ActorSystem
+import spray.json.DefaultJsonProtocol._
 import spray.json.JsObject
-import spray.json.JsValue
+import spray.json.pimpAny
+import spray.json.pimpString
+import whisk.common.ConsulClient
 import whisk.common.ConsulKV
 import whisk.common.ConsulKV.LoadBalancerKeys
 import whisk.common.Counter
 import whisk.common.Logging
-import scala.language.postfixOps
-
-
-class ActivationThrottle(consulServer: String,
-                         invokerHealth : InvokerHealth) extends Logging {
-
-    private val DEFAULT_NAMESPACE_CONCURRENCY_LIMITS_KEY = "whiskprops/DEFAULT_NAMESPACE_CONCURRENCY_LIMITS"
-    private val DEFAULT_LIMIT: BigInt = BigInt(100)
+import whisk.core.WhiskConfig
 
+/**
+ * Determines user limits and activation counts as seen by the invoker and the loadbalancer
+ * in a scheduled, repeating task for other services to get the cached information to be able
+ * to calculate and determine whether the namespace currently invoking a new action should
+ * be allowed to do so
+ *
+ * @param config containing the config information needed (consulServer)
+ */
+class ActivationThrottle(config: WhiskConfig)(
+    implicit val system: ActorSystem,
+    val ec: ExecutionContext) extends Logging {
+
+    val DEFAULT_NAMESPACE_CONCURRENCY_LIMITS_KEY = "whiskprops/DEFAULT_NAMESPACE_CONCURRENCY_LIMITS"
+    val DEFAULT_LIMIT = BigInt(100)
+
+    /**
+     * holds the values of the last run of the scheduler below to be gettable by outside
+     * services to be able to determine whether a namespace should be throttled or not based on
+     * the number of concurrent invocations it has in the system
+     */
+    private var userActivationCounter = Map[String, BigInt]()
```

Comment:
Yep i kept them as to not change the effects for the algorithms at all. You're right though that `Long` is probably sufficient, will change.